src/databricks/labs/ucx/framework/crawlers.py

```diff
@@ -151,7 +157,18 @@ def __init__(self):
 
     def execute(self, sql):
         logger.debug(f"[spark][execute] {sql}")
-        self._spark.sql(sql)
+        try:
+            immediate_response = self._spark.sql(sql)
+            status = immediate_response.status
```

Comment:
This is incorrect and too complicated. Don't involve statement status/etc. 

Keep it simple: Just use if ... in str(err): raise NotFound(...)